### DISCUSSION TITLE
jython HTTPRequest로 이미지파일을 전송하는 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by hoyongsh** [naver/ngrinder#822] [Q&A]
혹시 jython 스크립트로 이미지 업로드 테스트 할수 있는 방법은 있을까요?
resources 폴더안에 이미지 넣어 두고, 보낼때 PUT으로 보내려고 하는데 잘안되네요.

기존에는
multipart_header = [NVPair('Content-Type', 'multipart/form-data;boundary=--------adfadsf----')]
imgfiles = (NVPair("idcard_img_path","./resources/appIcon.png"),)
param = ( NVPair("run number", str(grinder.runNumber)), )
strBody = (NVPair("id",nickName)
data = Codecs.mpFormDataEncode(strBody, imgfiles, multipart_header)
result = request.POST("uri",data,multipart_header)

이런 방식으로 보냈었는데, img.png를 byte[]로 가지고 와서 바로
PUT(String uri, byte[] content, Map<String, String> headers)
이런방식으로 이미지를 보낼수 있는 방법이 있는지 궁금합니다.
현재 open("./resources/img.jpeg","rb") 로 이미지를 읽어오고 있는데 잘되지 않아서...

혹시 다른 이미지관련 라이브러리를 추가해야 되는지 궁금합니다.

**Comment by donggyu04** [comment]
신규 HttpClient를 사용해서 파일 업로드 테스트를 하시는 건가요?

테스트에 사용하실 스크립트와 똑같은 depth에 resources 폴더가 존재한다고 가정하고, resources/dummy.txt 가 있다고 가정하겠습니다.
아래 처럼 호출하시면 될 것 같습니다.

```python
from java.io import File
from org.ngrinder.http.multipart import MultipartEntityBuilder

...

multipart = MultipartEntityBuilder.create().addEntity("files", File("resources/dummy.txt")).build();
request.POST("{INPUT YOUR API URL}", multipart)
```

**Reply by hoyongsh** [reply]
@donggyu04 

신규 httpclient는 다음 프로젝트부터 적용이 가능할것 같아서, 현재는 기존에 있던 jython 그대로 사용중입니다.
기존 jython에서 사용을 하려면 위에 포함되어있는 라이브러리들을 수동으로 lib 폴더에 옮겨서 import 해서 사용하면 될까요?

**Reply by junoyoon** [reply]
과거 버전에 대한 사용 방법은 구글링을 하시면 많이 나올것 같습니다.
https://www.google.com/search?q=grinder+multipart+upload&rlz=1C5CHFA_enKR822KR822&oq=grinder+multipart+upload&aqs=chrome..69i57.5455j0j4&sourceid=chrome&ie=UTF-8

저희는 더이상 과거 버전에 대한 지원은 하지 않습니다.
